### DISCUSSION TITLE
[Feature] Upload multiple selected assets

### DISCUSSION THREAD

**Discussion by bigsinger** [immich-app/immich#3826] [Feature Request]
### The feature

The following instructions are for the mobile app: the current design requires you to click the cloud-shaped button to jump to the album selection, and then start the backup. This path is a bit long and inconvenient. You can choose whether you can display pictures on the home page of the App, and then there is a "Backup" button in the tool bar that prompts you. Click it directly to back up to the web server. This is the most convenient and efficient way. For the code layer, there is no need to compare all of them. For users, not all pictures need to be backed up. I need to choose by myself. In the current situation, I have to back up all of them to the cloud, and then delete the files that don't need to be backed up in the cloud, because otherwise it will cause a waste of storage space.

I just checked again and found that the "upload" button will appear when selecting a picture, but it will not appear when selecting in batches. I hope to add this function, it will be very convenient, thank you.

![image](https://github.com/immich-app/immich/assets/14847818/99e04ad9-c128-4e38-82ed-a56dc60aa248)

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by shenlong-tanwen** [comment]
> I just checked again and found that the "upload" button will appear when selecting a picture, but it will not appear when selecting in batches. I hope to add this function, it will be very convenient, thank you.

The "upload" button will appear if the selected assets includes only local assets. If it includes remote assets as well, the upload button will not appear.

> For users, not all pictures need to be backed up. I need to choose by myself. In the current situation, I have to back up all of them to the cloud, and then delete the files that don't need to be backed up in the cloud, because otherwise it will cause a waste of storage space.

You can already upload only selected assets to the cloud. However, the process is not trivial. It has to be updated to make it more approachable.

The current flow is such:
1) Enabled `"Show Storage indicator on asset tiles" under Settings -> Photo Grid` to make it easier to find local only assets.
2) In the app homepage, tap the cloud-shaped icon to jump to the Backup page. The albums selected here, are not only for the backup, but also displays the assets from the albums in the app timeline, irrespective of them being already uploaded to Immich or not. 
3) Once the required assets are selected, navigate back to the homepage which should now display both the local assets which are not backed up and remote assets which were already backed up. 
⚠️ You need not press the **Start Backup** button in the Backup page. That will upload all the assets in the selected albums.

4) You can now select local only assets from the timeline and use the upload button to selectively upload them to Immich

**Reply by bigsinger** [reply]
Will you plan to develop a more convenient way in the future?

**Reply by shenlong-tanwen** [reply]
Yes, Something has to be worked out to make this less confusing.

In my opinion, We can go with Google's approach of displaying all the assets on the device in the timeline always. Few might not like this, so maybe behind a toggle, which is enabled by default? Irrespective of the toggle, all the local albums will be displayed in the Library section under **Local Albums**. 

I'm not sure if there are a majority of users who wants to display only selective album assets in the timeline and library rather than displaying either all of them or none of them. In any case, this has to be discussed with the core team before finalizing a new approach.

**Reply by eygraber** [reply]
I don't think Google shows all of the assets on the device in the timeline always. For me they only show in the "Photos on device" section in the Library tab.

**Reply by bigsinger** [reply]
I don't really care about time, just give me the freedom to choose.

**Comment by bo0tzz** [comment]
This is now supported
